gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/jdbc/EmbedStatement.java

```diff
@@ -1202,6 +1203,11 @@ else if (SanityManager.DEBUG) {
               execFlags = (short) GemFireXDUtils.set(execFlags,
                   GenericStatement.QUERY_HDFS, true);
             }
+
+            if (routeQueryEnabled(cc)) {
```

Comment:
We can save this cost on rowstore mode. A check of isSnappyStore() should be good. Low latency queries will get saved from a pattern match.

Reply:
Ok